lib/mods_display/helpers/record_helper.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+module ModsDisplay
+  module Helpers
+    module RecordHelper
+      def display_content_field(field)
+        return unless field.respond_to?(:label, :values) && field.values.any?(&:present?)
+
+        display_content_label(field.label) + display_content_values(field.values)
+      end
+
+      def display_content_label(label)
+        content_tag :dt, label
+      end
+
+      def display_content_values(values)
+        values.map do |value|
+          content_tag :dd, value
+        end.join('').html_safe
+      end
+
+      def mods_display_label(label)
+        content_tag(:dt, label.delete(':')) + "\n".html_safe
+      end
+
+      def mods_display_content(values, delimiter = nil)
+        if delimiter
+          content_tag(:dd, values.map do |value|
+            link_urls_and_email(value) if value.present?
+          end.compact.join(delimiter).html_safe)
+        else
+          Array[values].flatten.map do |value|
+            content_tag(:dd, link_urls_and_email(value.to_s).html_safe) if value.present?
+          end.join.html_safe
+        end
+      end
+
+      def mods_record_field(field, delimiter = nil)
+        return unless field.respond_to?(:label, :values) && field.values.any?(&:present?)
+
+        mods_display_label(field.label) + mods_display_content(field.values, delimiter)
+      end
+
+      def mods_name_field(field, &block)
```

Comment:
The pattern here of just passing the block around seemed to be the only way I knew how to get the desired `link_to` functionality. See some of the tests for examples:

```ruby
      it 'should join the label and values' do
        name = mods_name_field(name_field) do |name|
          link_to(name, searches_path(q: "\"#{name}\"", search_field: 'search_author'))
        end
        expect(name).to match /<dt>Contributor<\/dt>/
        expect(name).to match /<dd><a href.*<\/dd>/
      end
```

This is another change we will need to make in SearchWorks.